Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsActivity.java

```diff
@@ -172,22 +189,15 @@ protected void onActivityResult(int requestCode, int resultCode, Intent result)
   {
     super.onActivityResult(requestCode, resultCode, result);
 
-    // Save modified non-FilePicker settings beforehand since finish() won't save them.
-    // onStop() must come before handling the resultCode to properly save FilePicker selection.
-    mPresenter.onStop(true);
-
     // If the user picked a file, as opposed to just backing out.
     if (resultCode == MainActivity.RESULT_OK)
     {
       mPresenter.onFileConfirmed(FileBrowserHelper.getSelectedPath(result));
-
-      // Prevent duplicate Toasts.
-      if (!mPresenter.shouldSave())
-      {
-        Toast.makeText(this, "Saved settings to INI files", Toast.LENGTH_SHORT).show();
-      }
     }
 
+    // Save modified settings since finish() won't save them.
+    mPresenter.onStop(true);
+
     // TODO: After result of FilePicker, duplicate SettingsActivity appears.
     //       Finish to avoid this. Is there a better method?
     finish();
```

Comment:
Now that I think about it, we'd both need to do less merge conflict resolution if turned #9014 into a review comment. Please remove `mPresenter.onStop(true);` and `finish();` after rebasing.

Reply:
I'd rather keep PR #9014 open and solve the conflict once that PR is merged, since I don't know how long it's going to take for this PR to get reviewed and merged, but if you say so...

Done in the latest push.

Reply:
I started pulling code from this PR to make `FilePickerViewHolder.bind` work the same way but those changes sprawled across multiple files. I think the change makes sense here since it relies on part of this PR.